packages/typespec-ts/src/modular/buildClientContext.ts

```diff
@@ -63,77 +69,100 @@ export function buildClientContext(
       }),
     docs: ["Optional parameters for the client."]
   });
-  if (isRLCMultiEndpoint(dpgContext)) {
-    clientContextFile.addImportDeclaration({
-      moduleSpecifier: `../../rest/${subfolder}/index.js`,
-      namedImports: [`Client`]
-    });
 
-    clientContextFile.addExportDeclaration({
-      moduleSpecifier: `../../rest/${subfolder}/index.js`,
-      namedExports: [`Client`]
-    });
-    factoryFunction = clientContextFile.addFunction({
-      docs: getDocsFromDescription(description),
-      name: `create${name}`,
-      returnType: `Client.${client.name}`,
-      parameters: params,
-      isExported: true
-    });
-  } else {
-    const rlcClientName = client.rlcClientName;
+  // TODO use binder here
+  // (for now) now logger for unbranded pkgs
+  if (codeModel.options.flavor === "azure") {
     clientContextFile.addImportDeclaration({
-      moduleSpecifier: `${
-        subfolder && subfolder !== "" ? "../" : ""
-      }../rest/index.js`,
-      namedImports: [`${rlcClientName}`]
+      moduleSpecifier:
+        codeModel.clients.length > 1 ? "../../logger.js" : "../logger.js",
+      namedImports: ["logger"]
     });
+  }
 
-    clientContextFile.addExportDeclaration({
-      moduleSpecifier: `${
-        subfolder && subfolder !== "" ? "../" : ""
-      }../rest/index.js`,
-      namedExports: [`${rlcClientName}`]
-    });
+  const factoryFunction = clientContextFile.addFunction({
+    docs: getDocsFromDescription(description),
+    name: `create${name}`,
+    returnType: `${client.rlcClientName}`,
+    parameters: params,
+    isExported: true
+  });
 
-    factoryFunction = clientContextFile.addFunction({
-      docs: getDocsFromDescription(description),
-      name: `create${name}`,
-      returnType: `${rlcClientName}`,
-      parameters: params,
-      isExported: true
+  const endpointParam = buildGetClientEndpointParam(factoryFunction, client);
```

Comment:
feel like we might have missed the default logic handling for other parameters

Reply:
https://github.com/timovv/autorest.typescript/blob/62159e3252799f32b10629a79dcb3d2552fd219d/packages/rlc-common/src/buildClient.ts#L319-L347

Reply:
I think this is covered here but let me know if there's something missing in that logic:

https://github.com/timovv/autorest.typescript/blob/62159e3252799f32b10629a79dcb3d2552fd219d/packages/typespec-ts/src/modular/helpers/clientHelpers.ts#L79-L93

Reply:
I did not see the default value was handled in https://github.com/Azure/autorest.typescript/blob/dd505de1900770cb747f3fb73bfc11500c58ae5d/packages/typespec-test/test/parametrizedHost/generated/typespec-ts/src/rest/parametrizedHostClient.ts#L30-L32 and currently it's missing in https://github.com/timovv/autorest.typescript/blob/62159e3252799f32b10629a79dcb3d2552fd219d/packages/typespec-test/test/parametrizedHost/generated/typespec-ts/src/api/parametrizedHostContext.ts#L22-L25

Reply:
I also think this kind of default value change can not be catched by the api view.md @joheredi this is a potential risk if we just check in the api view.md file.

Reply:
Thanks. This was happening because we weren't populating the `clientDefaultValue` for these params in the code model, like I was expecting. I updated the code model to set the `clientDefaultValue` to the default value for endpoint path params only, and this fixed the issue. I think this is the right behavior, it doesn't make sense to have a server-side default value for the endpoint.

> I also think this kind of default value change can not be catched by the api view.md

This is true, but it also would have been caught by an integration test (or a scenario test like I was showing yesterday), so maybe we can add coverage there.

Reply:
I don't think this is correct, we should respect the parameter default in the `@server` decorators, that's why we should respect the endpoint default value, also in this case, those parameters are parts of the endpoint https://github.com/Azure/autorest.typescript/blob/dd505de1900770cb747f3fb73bfc11500c58ae5d/packages/typespec-test/test/parametrizedHost/generated/typespec-ts/src/rest/parametrizedHostClient.ts#L30-L36 

In this case, if we don't populating the clientDefaultValue, those parameters should be required and go into the getClient method signature instead of options.

Reply:
Can you give an example of what this example should look like?

Reply:
Is the issue here that `apiVersion` isn't getting a default value when it should be? Or is there something else that is also not being considered?

Reply:
This is the typespec definition https://github.com/Azure/autorest.typescript/blob/dd505de1900770cb747f3fb73bfc11500c58ae5d/packages/typespec-test/test/parametrizedHost/spec/main.tsp#L20-L32 , we ignore clientDefaultValue for everywhere else but this `@server` decorator. 

if we choose to ignore the default value for it, then we should not handle the default value for endpoint or apiVersion either. the `createParameterizedHost` function should be like 
```ts
export function createParametrizedHost(
  credential: TokenCredential,
  host: string,
  domain: string,
  suffix: string,
  apiVersion: string,
  options: ParametrizedHostClientOptionalParams = {},
): 
```

There're some more context here for how to handle endpoint and how to handle the clientDefaultValue 
https://github.com/Azure/autorest.typescript/pull/2032#discussion_r1349875580 
https://github.com/Azure/autorest.typescript/issues/2049

Reply:
> Is the issue here that `apiVersion` isn't getting a default value when it should be? Or is there something else that is also not being considered?

we have the default value for apiVersion https://github.com/Azure/autorest.typescript/pull/2561/files#diff-c381779e571bb42446df22900b2c16f1f7036a24f71042ac0bd83e4ef47ff630R28